### PR TITLE
AC: Fix coco_orig_metric

### PR DESCRIPTION
**Pull request by Anna-Mironova**
Calculation of coco_orig_metric has been fixed.   AP has been reduced to 66.6% from 69%. After fix AP for mentioned model is 68.37%. 

In this PR usage of original and generated annotation file has been combined by condition. If there is annotation file, it is used when calculating the metric. If annotation file is missing or there is subsample_size option, use generated annotation file.

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py [hunk 1]
```diff
@@ -252,9 +333,32 @@ def _run_coco_evaluation(coco, coco_res, iou_type='bbox', threshold=None):
...
+                    return True
+                else:
+                    return False
+            else:
+                return False
+
     def compute_precision_recall(self, annotations, predictions):
-        coco = self._prepare_coco_structures(annotations)
-        coco_data_to_store = self._convert_data_to_coco_format(predictions)
+        if(self._use_original_coco()):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py:352]:
```suggestion
        if self._use_original_coco():
```

File: tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py [hunk 2]
```diff
@@ -252,9 +333,32 @@ def _run_coco_evaluation(coco, coco_res, iou_type='bbox', threshold=None):
...
+            coco, map_coco_img_file_name_to_img_id, map_pred_label_id_to_coco_cat_id = \
+                self._prepare_original_coco_structures()
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py:354]:
```suggestion
            structures = self._prepare_original_coco_structures()
            coco, map_coco_img_file_name_to_img_id, map_pred_label_id_to_coco_cat_id = structures
                
```

File: tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py [hunk 3]
```diff
@@ -107,7 +108,56 @@ def _prepare_coco_structures(self, annotations):
...
+
+    def _prepare_original_coco_structures(self):
+        annotation_conversion_parameters = self.dataset.config.get('annotation_conversion')
+        annotation_file = annotation_conversion_parameters.get('annotation_file')
+        has_background = annotation_conversion_parameters.get('has_background', False)
+        use_full_label_map = annotation_conversion_parameters.get('use_full_label_map', False)
+
+        meta = self.dataset.metadata
+
+        if COCO is None:
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py:128]: If I right remember we move this check to configure, is not it?
If yes, possibly we do not need to check it twice

Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py:128]: there are checks for self.dataset.metadata and self.dataset.metadata.get('label_map') in configure. I can move check for COCO in configure too

**Pull request merged**